### DISCUSSION TITLE
Compare at price not available for discount apis

### DISCUSSION THREAD

**Discussion by Will-mexi** [Shopify/function-examples#551] [Feedback: Discounts APIs]
Hello,
I have been working on migrating our scripts over to functions. We use a lot of sales that work by setting the compare at price to the current price of the variant and then setting the price of a variant to a discounted amount. 

I would like to filter out these variants from the automatic evergreen discounts that run as a function but am unable to get the compare at price of a productvariant in my input query. Is there an easy way to do this or is it possible to add the compare at price of a variant to the input query on shopify's end? 
Thanks!

**Comment by davejcameron** [comment]
Here is a sample input query that gets compare at price
```
query Input {
  cart {
    lines {
      cost {
        amountPerQuantity {
          amount
        }
        compareAtAmountPerQuantity {
          amount
        }
      }
      merchandise {
        __typename
        ... on ProductVariant {
          id
        }
      }
    }
  }
  discountNode {
    metafield(namespace: "$app:config", key: "function-config") {
      value
    }
  }
}
```

**Reply by Will-mexi** [reply]
That solved it thank you!
